cli/internal/terraform/loader.go

```diff
@@ -27,11 +28,26 @@ var ErrTerraformWorkspaceDifferentFiles = errors.New("creating cluster: trying t
 //go:embed terraform/*/.terraform.lock.hcl
 var terraformFS embed.FS
 
+const (
+	IAMAction = iota
+	ClusterAction
+)
+
 // prepareWorkspace loads the embedded Terraform files,
 // and writes them into the workspace.
-func prepareWorkspace(fileHandler file.Handler, provider cloudprovider.Provider, workingDir string) error {
-	// use path.Join to ensure no forward slashes are used to read the embedded FS
-	rootDir := path.Join("terraform", strings.ToLower(provider.String()))
+func prepareWorkspace(action int, fileHandler file.Handler, provider cloudprovider.Provider, workingDir string) error {
+	var rootDir string
+	// select the correct path for the Terraform files depending on the action
+	switch action {
+	case IAMAction:
+		// use path.Join to ensure no forward slashes are used to read the embedded FS
+		rootDir = path.Join("terraform", strings.ToLower(provider.String()), "iam")
+	case ClusterAction:
+		// use path.Join to ensure no forward slashes are used to read the embedded FS
+		rootDir = path.Join("terraform", strings.ToLower(provider.String()))
```

Comment:
```suggestion
	case IAMAction:
		// use path.Join to ensure no forward slashes are used to read the embedded FS
		rootDir = path.Join("terraform", "iam", strings.ToLower(provider.String()))
	case ClusterAction:
		// use path.Join to ensure no forward slashes are used to read the embedded FS
		rootDir = path.Join("terraform", strings.ToLower(provider.String()))
```

Would suggest changing terraform/gcp/iam -> terraform/iam/gcp so we don't necessarily extract it for the cluster creation.